src/components/motor/MotorController.cpp

```diff
@@ -47,6 +70,22 @@ void MotorController::StopRinging() {
   nrf_gpio_pin_set(pinMotor);
 }
 
-void MotorController::StopMotor(void* p_context) {
-  nrf_gpio_pin_set(pinMotor);
+
+void MotorController::Vibrate(void* p_context) {
+  
+  if (step >= tunes[runningTune].length || step >= 8) { //end of tune turn off vibration
+    nrf_gpio_pin_set(pinMotor);
+    return; 
+  }  
+ 
+  if (((1 << step) & tunes[runningTune].tune) > 0) {
+    nrf_gpio_pin_clear(pinMotor);
+  } else {
+    nrf_gpio_pin_set(pinMotor);
+  }
```

Comment:
Any reason not to use RunForDuration() for the patterns as well? It would provide better abstraction.

Reply:
RunForDuration() is public facing API call, that is currently implemented as "just another tune to play". Would like to not add any more logic to the public facing API. Adding enableMotor() and disableMotor() calls would maybe make it more readable.

Reply:
>RunForDuration() is public facing API call, that is currently implemented as "just another tune to play".

That's because you changed it that way. That function didn't necessarily have to be changed at all.

Reply:
Refactored the approach to use a unified ScheduleVibrationTimer call.